src/libs/metadata.c

```diff
@@ -296,18 +285,164 @@ int position()
+{
+  const dt_lib_metadata_t *d = (dt_lib_metadata_t *)self->data;
   /* lets trigger an expose for a redraw of widget */
   if(d->editing)
   {
-    write_metadata(user_data);
+    write_metadata(self);
     gtk_window_set_focus(GTK_WINDOW(dt_ui_main_window(darktable.gui->ui)), NULL);
   }
-  update(user_data, FALSE);
+  update(self, FALSE);
+}
+
+static gboolean _metadata_list_size_changed(GtkWidget *window, GdkEvent  *event, GtkCellRenderer *renderer)
+{
+  GdkEventConfigure *data = (GdkEventConfigure *)event;
+
+//  GtkTreeView *listview = g_object_get_qdata(G_OBJECT(renderer), g_quark_from_static_string("listview"));
+  g_object_set(G_OBJECT(renderer), "wrap-width", data->width, NULL);
+  // TODO make it works. renderer should take in account the new wrap-width and calculate a new record height
+  gtk_widget_queue_draw(GTK_WIDGET(window));
+  return FALSE;
+}
+
+typedef struct dt_lib_metadata_dialog_t
+{
+  GtkTextView *textview;
+  GtkTreeView *listview;
+  GtkDialog *dialog;
+} dt_lib_metadata_dialog_t;
+
+static gboolean metadata_selected(GtkWidget *listview, GdkEventButton *event, dt_lib_metadata_dialog_t *d)
+{
+  if(event->type == GDK_BUTTON_PRESS && event->button == 1)
+  {
+    GtkTreePath *path = NULL;
+    // Get tree path for row that was clicked
+    if(gtk_tree_view_get_path_at_pos(GTK_TREE_VIEW(listview), (gint)event->x, (gint)event->y, &path, NULL, NULL, NULL))
+    {
+      GtkTreeModel *liststore = gtk_tree_view_get_model(GTK_TREE_VIEW(d->listview));
+      GtkTreeIter iter;
+      if(gtk_tree_model_get_iter(liststore, &iter, path))
+      {
+        gchar *text;
+        gtk_tree_model_get(liststore, &iter, 0, &text, -1);
+        GtkTextBuffer *buffer = gtk_text_view_get_buffer(d->textview);
+        gtk_text_buffer_set_text(buffer, text, -1);
+        g_free(text);
+        gtk_tree_path_free(path);
+        gtk_dialog_response(d->dialog, GTK_RESPONSE_YES);
+        return TRUE;
+      }
+    }
+    gtk_tree_path_free(path);
+  }
+  gtk_dialog_response(d->dialog, GTK_RESPONSE_NONE);
+  return FALSE;
+}
+
+static void config_button_clicked(GtkButton *button, dt_lib_module_t *self)
+{
+  GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("metadata settings"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
+                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_YES, NULL);
+  GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
+
+  GtkWidget *grid = gtk_grid_new();
+  gtk_container_add(GTK_CONTAINER(area), grid);
+  GtkWidget *label = gtk_label_new(_("metadata"));
+  gtk_grid_attach(GTK_GRID(grid), label, 0, 0, 1, 1);
+  label = gtk_label_new(_("hidden"));
+  gtk_grid_attach(GTK_GRID(grid), label, 1, 0, 1, 1);
+  label = gtk_label_new(_("private"));
+  gtk_grid_attach(GTK_GRID(grid), label, 2, 0, 1, 1);
+  label = gtk_label_new(_("height"));
```

Comment:
What is this height ? I have it on my dialog but there is no check box associated ? It does not appears on your screenshot BTW.

Reply:
I have forgotten this.Must be removed. It's now done by ctrl-wheel.